File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    lost_set = set(lost) - set(reserve)
    reserve_set = set(reserve) - set(lost)
    student = [1] * (n + 1)
    for l in lost_set:
        if (l - 1) in reserve_set:
            reserve_set.remove(l-1)
            continue
        if (l + 1) in reserve_set:
            reserve_set.remove(l+1)
            continue
        student[l] = 0
    
    return student.count(1) - 1

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    const lostSet = new Set(lost)
    const reserveSet = new Set(reserve)
    
    const realLostSet = new Set([...lostSet].filter(el => {
        if (reserveSet.has(el)) {
            reserveSet.delete(el)
            return false
        }
        return true
    }))
    
    for (let i of [...reserveSet].sort((a, b) => a - b)) {
        if (realLostSet.has(i - 1)) {
            realLostSet.delete(i - 1)
            continue
        }
        if (realLostSet.has(i + 1)) {
            realLostSet.delete(i + 1)
            continue
        }
    }
    answer = n - realLostSet.size
    return answer
}